boot/freeldr/freeldr/arch/i386/multiboot.S

```diff
@@ -173,23 +173,33 @@ mb3:
     mov ss, ax
 
     /* Jump to relocated code */
-    ljmp HEX(08), mb4
+    .byte HEX(EA)
+    .long mb4
+    .word HEX(08)
 
 mbfail:
     int 3
 mbstop:
     jmp short mbstop  /* We should never get here */
-
-.code16
 mb4:
     /* Disable protected mode */
+#ifdef _M_IX86
     mov eax, cr0
+#else
+    mov rax, cr0
+#endif
+    .byte HEX(66)
     and eax, CR0_PE_CLR
+#ifdef _M_IX86
     mov cr0, eax
+#else
+    mov cr0, rax
+#endif
 
     /* Jump to real entry point */
-    ljmp16 0, FREELDR_BASE
-.endcode16
+    .byte HEX(EA)
+    .word FREELDR_BASE
+    .word HEX(00)
```

Comment:
Why?

Reply:
but the problem is the ljmp16 and ljmp doesn't exist in Amd64 GCC and Clang compilers

Reply:
That is why we have asm.inc. To define these as macros.

Reply:
i modified the asm.inc to resolve the ljmp and ljmp16 problem